UnityProject/Assets/Scripts/UI/Systems/MainHUD/PlayerHealthUI.cs

```diff
@@ -193,15 +193,33 @@ public void SetBodyTypeOverlay(BodyPart bodyPart)
 						damageColor = disabledBodyPartColor;
 						break;
 					case DamageSeverity.Max:
+						bodyPartColor = damageMonitorColors[6];
+						damageColor = destroyedBodyPartColor;
+						break;
 					default:
 						bodyPartColor = damageMonitorColors[6];
 						damageColor = destroyedBodyPartColor;
 						break;
 				}
-
-				bodyPartListeners[i].SetDamageColor(damageColor);
-				bodyPartListeners[i].SetBodyPartColor(bodyPartColor);
+				if (IsLocalPlayer(bodyPart))
+				{
+					bodyPartListeners[i].SetDamageColor(damageColor);
+					bodyPartListeners[i].SetBodyPartColor(bodyPartColor);
+				}
+				else
+				{
+					if (bodyPart.healthMaster != null)
+					{
+						HealthDollIpdate.SendTo(i, damageColor, bodyPartColor, bodyPart.healthMaster.gameObject);
+					}
```

Comment:
surely this wont work as the server will only wont have multiple PlayerHealthUI scripts, shouldnt the damage be synced in the PlayerHealthV2

Reply:
it's good point since it will only iterate over what server has but the client is worked out from the body part health master

Reply:
could you give me a good idea of how to make it a synchvar instead

Reply:
maybe a synclist or syncdictionary of all the BodyPartType and their current health, then on client in the hook you sent that dictionary to PlayerHealthUI where the list is iterated and then the bodyPartType is compared to the health doll, then the corresponding damage of the part is shown?

Reply:
didn't know about HealthStateController nice